iroha/src/isi.rs

```diff
@@ -17,106 +19,56 @@ impl Id {
-        println!("World state: {:?}", world_state_view.world);
-        match self.command_type {
-            1 => {
-                let instruction: AddAssetQuantity = self.payload.clone().into();
-                world_state_view
-                    .world
-                    .account(&instruction.account_id)
-                    .unwrap()
-                    .assets
-                    .insert(
-                        instruction.asset_id.clone(),
-                        Asset::new(instruction.asset_id),
-                    );
-            }
-            5 => {
-                let instruction: CreateAccount = self.payload.clone().into();
-                world_state_view
-                    .world
-                    .domain(&instruction.domain_name)
-                    .unwrap()
-                    .accounts
-                    .insert(
-                        instruction.account_id.clone(),
-                        Account::new(instruction.account_id),
-                    );
-            }
-            7 => {
-                let instruction: CreateDomain = self.payload.clone().into();
-                world_state_view
-                    .world
-                    .add_domain(Domain::new(instruction.domain_name));
-            }
-            17 => {
-                let instruction: TransferAsset = self.payload.clone().into();
-                let asset = world_state_view
-                    .world
-                    .account(&instruction.source_account_id)
-                    .unwrap()
-                    .assets
-                    .remove(&instruction.asset_id)
-                    .unwrap();
-                world_state_view
-                    .world
-                    .account(&instruction.destination_account_id)
-                    .unwrap()
-                    .assets
-                    .insert(instruction.asset_id.clone(), asset);
-            }
-            _ => (),
+///
+#[derive(Clone, Debug, PartialEq, Encode, Decode)]
+pub enum Contract {
+    AddSignatory(account::isi::AddSignatory),
+    AppendRole(account::isi::AppendRole),
+    CreateAccount(account::isi::CreateAccount),
+    CreateRole(account::isi::CreateRole),
+    AddAssetQuantity(asset::isi::AddAssetQuantity),
+    TransferAsset(asset::isi::TransferAsset),
+    CreateAsset(asset::isi::CreateAsset),
+    CreateDomain(domain::isi::CreateDomain),
+    AddPeer(peer::isi::AddPeer),
+}
+
+impl Contract {
+    pub fn invoke(&self, world_state_view: &mut WorldStateView) -> Result<(), String> {
+        use Contract::*;
+        match self {
+            AddAssetQuantity(instruction) => instruction.execute(world_state_view),
+            CreateAccount(instruction) => instruction.execute(world_state_view),
+            CreateDomain(instruction) => instruction.execute(world_state_view),
+            TransferAsset(instruction) => instruction.execute(world_state_view),
+            _ => Err("Not supported.".to_string()),
```

Comment:
Should probably make error handling more specific

Reply:
@takemiyamakoto like "{} Instruction is not supported.", instruction ?